### PR TITLE
Extended page objects to PageObject class.

### PR DESCRIPTION
**Pull request by nesibfisohmh**
Added method to switch between frames.
Created flow for view products for specified school.

### CODE REVIEW

**Comment by nesibfisohmh** [comment]
Please review

File: src/test/java/com/hmhco/hmof/pages/Products/EligibleProductsPage.java [hunk 1]
```diff
@@ -1,7 +1,30 @@
...
- */
-public class EligibleProductsPage {
+import net.serenitybdd.core.pages.PageObject;
+import net.serenitybdd.core.pages.WebElementFacade;
+import org.openqa.selenium.support.FindBy;
+
+public class EligibleProductsPage  extends PageObject {
+
+    // locators
+    @FindBy(id = "caption-0-7-0")
```
Comment (dgracehmh) [inline src/test/java/com/hmhco/hmof/pages/Products/EligibleProductsPage.java]: static?

File: src/test/java/com/hmhco/hmof/pages/Products/ActivateClassProductsPage.java [hunk 2]
```diff
@@ -1,7 +1,83 @@
...
+
+    @FindBy(id = "search")
+    public static WebElementFacade button_search;
+
+    @FindBy(css = "input[type='Reset']")
+    public static WebElementFacade button_reset;
+
+
+    // methods
+    public void filterBySubject() throws Throwable {
```
Comment (dgracehmh) [inline src/test/java/com/hmhco/hmof/pages/Products/ActivateClassProductsPage.java]: I dont know the page but maybe one method is all thats needed, filterBy(FilterMethodEnum filterMethodEnum)
Where you create an enum with all the filter methods and handle it that way. What do you reckon?

Comment (dgracehmh) [inline src/test/java/com/hmhco/hmof/pages/Products/ActivateClassProductsPage.java]: btw. i'm crap at naming things, feel free to use something better than FilterMethodEnum :)

File: src/test/java/com/hmhco/hmof/pages/Products/SearchProductAllocationsPage.java [hunk 3]
```diff
@@ -1,7 +1,30 @@
...
- */
-public class SearchProductAllocationsPage {
+import net.serenitybdd.core.pages.PageObject;
+import net.serenitybdd.core.pages.WebElementFacade;
+import org.openqa.selenium.support.FindBy;
+
+public class SearchProductAllocationsPage extends PageObject {
+
+    // locators
+    @FindBy(id = "caption-0-7-3")
```
Comment (marijanaj) [inline src/test/java/com/hmhco/hmof/pages/Products/SearchProductAllocationsPage.java]: I think it is static.

File: src/test/java/com/hmhco/hmof/steps/RegressionDA/ProductManagement.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    ProductManagementPageSteps productManagementPageSteps;
+
+    @Given("the DA view school products '(.*)'")
+    public void the_DA_view_school_products_testSchoolProduct(String schoolProduct) throws Throwable {
+        productManagementPageSteps.viewSchoolProducts(schoolProduct);
+    }
+
+    @Given("the DA search school products '(.*)', '(.*)'")
+    public void the_DA_search_school_products_testSchoolProduct(){
+
```
Comment (nesibfisohmh) [inline src/test/java/com/hmhco/hmof/steps/RegressionDA/ProductManagement.java]: sure, is it fine to add something like:
    @Given("the DA search school products '(.*)', '(.*)'")
    public void the_DA_search_school_products_testSchoolProduct(){
		Assert.assertNotNull("", true);
	}

File: src/test/java/com/hmhco/hmof/pages/Products/SearchProductAllocationsPage.java (hunk 3 above)
Comment (nesibfisohmh) [inline src/test/java/com/hmhco/hmof/pages/Products/SearchProductAllocationsPage.java]: I would say it is static as well for each of these weird ids...

File: src/test/java/com/hmhco/hmof/pages/Products/DeleteProductAllocationsPage.java [hunk 5]
```diff
@@ -1,7 +1,28 @@
...
- */
-public class DeleteProductAllocationsPage {
+import net.serenitybdd.core.pages.PageObject;
+import net.serenitybdd.core.pages.WebElementFacade;
+import org.openqa.selenium.support.FindBy;
+
+public class DeleteProductAllocationsPage extends PageObject {
+
+    // locators
+    @FindBy(id = "caption-0-7-2")
```
Comment (nesibfisohmh) [inline src/test/java/com/hmhco/hmof/pages/Products/DeleteProductAllocationsPage.java]: I would say so

File: src/test/java/com/hmhco/hmof/pages/Products/EligibleProductsPage.java (hunk 1 above)
Comment (nesibfisohmh) [inline src/test/java/com/hmhco/hmof/pages/Products/EligibleProductsPage.java]: I would say so

File: src/test/java/com/hmhco/hmof/pages/Products/ActivateClassProductsPage.java (hunk 2 above)
Comment (nesibfisohmh) [inline src/test/java/com/hmhco/hmof/pages/Products/ActivateClassProductsPage.java]: Will switch to enum approach. It does seems better.

**Comment by nesibfisohmh** [comment]
I'll close this PR and merge it to newly open branch so we can wait for Darragh approve with latest updates on master branch.

**Pull request closed without merging**